src/compile/data/impute.ts

```diff
@@ -23,16 +26,59 @@ export class ImputeTransformNode extends DataFlowNode {
     return {signal: `sequence(${start}, ${stop}, ${step})`};
   }
 
-  public assemble(): VgImputeTransform {
-    const {impute, key, keyvals, method, groupby, value} = this.transform;
-    return {
-      type: 'impute',
-      field: impute,
-      key,
-      ...(keyvals ? {keyvals: isImputeSequence(keyvals) ? this.processSequence(keyvals) : keyvals}: {}),
-      ...(method ? {method}: {}),
-      ...(groupby ? {groupby}: {}),
-      ...(value ? {value}: {})
-    };
+  public static makeFromTransform(parent: DataFlowNode, imputeTransform: ImputeTransform): ImputeTransformNode {
+    return new ImputeTransformNode (parent, imputeTransform);
+  }
+
+  public static makeFromEncoding(parent: DataFlowNode, model: UnitModel) {
+    const transformRepresentation = normalizeimputeProperties(model.encoding);
+    if (transformRepresentation === undefined) {
+      return null;
+    }
+    return new ImputeTransformNode(parent, transformRepresentation);
+  }
+
+  public assemble(): [VgImputeTransform] | [VgImputeTransform, VgWindowTransform, VgFormulaTransform] {
+    const {impute, key, keyvals, method, groupby, value, frame} = this.transform;
+    if (method !== 'value' && frame !== undefined) {
+      const initialImpute: VgImputeTransform = {
```

Comment:
Please confirm if you have played with this in Vega to see if we really need additonal impute.

Reply:
Do you mean having the same `VgImputeTransform` for both `frame` and regular imputes and then the 2 optional transforms if `frame` is present?

If so then I thought this was easier to follow, since sharing the `VgImputeTransform` leads to adding 3 or 4 more checks.(eg. when `method === 'value' && frame !== undefined`)

Reply:
I mean, I don't know if we really need an impute at all if frame is presensted. (We might need, but you better try and tell me why.)

Reply:
Thinking more about it I still don't like the fact that encoding impute with frame doesn't behave like encoding's impute without frame.  Perhaps we should use `undefined` as the placeholder instead of null?

Reply:
If I understand correctly, you are not comfortable with `impute` altering the values of data objects that were present in the `source` with `null` values?
Other than using `undefined`, I can only think of using `Symbol` to generate a unique value to initially `impute` with.

Reply:
Yes, if impute affect prior `null` values, it should always affect them (not only when `frame` is specified).

Symbol may work too.  Give it a try?

Reply:
`Symbol` cannot be used in this case since the  `value` we are imputing with has to be converted to a `string` for use in the `formula` step.
Other than generating a random string, I think `undefined` is the only option.